src/ssl.c

```diff
@@ -13314,16 +13314,18 @@ int wolfSSL_EVP_MD_type(const WOLFSSL_EVP_MD *md)
             WOLFSSL_MSG("no type set");
             return 0;   /* failure */
         }
-        ctx->bufUsed = 0;
-        ctx->lastUsed = 0;
-        ctx->flags   = 0;
+        if (ctx->cipherType == 0xff){
+            ctx->bufUsed = 0;
+            ctx->lastUsed = 0;
+            ctx->flags   = 0;
+        }
         ret = 0;
 #ifndef NO_AES
         if (ctx->cipherType == AES_128_CBC_TYPE ||
             (type && XSTRNCMP(type, EVP_AES_128_CBC, EVP_AES_SIZE) == 0)) {
             WOLFSSL_MSG("EVP_AES_128_CBC");
             ctx->cipherType = AES_128_CBC_TYPE;
-            ctx->flags      = WOLFSSL_EVP_CIPH_CBC_MODE;
+            ctx->flags     |= WOLFSSL_EVP_CIPH_CBC_MODE;
```

Comment:
Technically you'll need to clear the previous mode for each of these before setting it. That would look like `ctx->flags &= ~WOLFSSL_EVP_CIPH_MODE`. This takes 0x7 and inverts it to 0xF8, so the operation is `ctx->flags &= 0xF8`, which clears the lower bits and leaves the upper ones set.

Reply:
Thank you for your suggestion.
I added code to clear lower 3 bit before setting encrypt mode flags.